healthcheck/healthcheck.py

```diff
@@ -181,3 +182,27 @@ def _assess_overall_status(self):
             return False
 
         return True
+
+
+def _file_exists(path):
+    """Return True if a file exists at `path` (even if it can't be read),
+    otherwise False.
+
+    This is different from os.path.isfile and os.path.exists which return
+    False if a file exists but the user doesn't have permission to read it.
+    """
+    try:
+        os.stat(path)
+        return True
+    except OSError as e:
+
```

Comment:
> Two edge cases, both look bad to me

I don't follow, what do you suggest?

> And remove comment before raise, it really says nothing.

As it is now you can read the comments and understand what is happening without even looking at the code, which I think is much easier on the eyes (which can skip over very quickly) and the brain. You don't have to spend any time wondering what `errno.EACCES` and `errno.ENOENT` mean. You don't have to decrypt the control flow and see that it raises an exception because the two if conditions failed and so the returns haven't happened and so that represents a case which the programmer wasn't aware of and accounted for with a fail fast option but which really probably doesn't exist. I think the comment says a lot.